### PR TITLE
Feature/endpoint

### PR DESCRIPTION
**Pull request by edtu1**
Addition of endpoint (Still needs to be ironed out with naming and how the response is returned) + updated headings to match other specs.

### CODE REVIEW

**Comment by agreenhalgh** [comment]
Need to move to use a fhir operation, needs to suport the same search & retrieval parameters as in ARS. OAS should not privide ecxamples, just link through to simplifier.

File: specification/gp-connect-access-record-fhir.yaml [hunk 1]
```diff
@@ -33,79 +40,105 @@ info:
...
+    
+    * [User-restricted RESTful APIs - NHS login separate authentication and authorisation](https://digital.nhs.uk/developer/guides-and-documentation/security-and-authorisation/user-restricted-restful-apis-nhs-login-separate-authentication-and-authorisation)
 
-      ## Environments and testing
+    ## Environments and testing
 
     | Purpose          | URL                                                   |
     | -------          | ---                                                   |
-    | Sandbox          | [https://sandbox.api.service.nhs.uk/gp-connect-get-structured-record-fhir](https://sandbox.api.service.nhs.uk/gp-connect-get-structured-record-fhir) |
+    | Sandbox          | [https://sandbox.api.service.nhs.uk/gp-connect/patient-facing/get-structured-record/](https://sandbox.api.service.nhs.uk/gp-connect/patient-facing/get-structured-record) |
```
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:139]: This isn't consistent with the servers section on line 209-215

File: specification/gp-connect-access-record-fhir.yaml [hunk 2]
```diff
@@ -213,4 +253,119 @@ servers:
     description: Integration test environment.
   - url: 'https://api.service.nhs.uk/gp-connect-access-record-fhir'
     description: Production environment.
-paths: {}
+paths: 
+  /FHIR/STU3/$access-record-structured:
```
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:255]: Not sure this is consistent with the servers section above and the environments section

File: specification/gp-connect-access-record-fhir.yaml [hunk 3]
```diff
@@ -115,54 +148,52 @@ info:
...
 
+    ### Consumer assurance process
 
+    On approval of a use case, we will support you through the assurance process through to go live. We will discuss the assurance process and artefacts with you to help you understand the requirements.
 
-    ## Consumer assurance process
+    Start your development work within 6 months of use case approval. If you miss this date, a review or new submission of the use case will be required. Changes or additional development will also require a review or new use case submission. For full details of the technical conformance process, see GP Connect Consumer Supplier Test Assurance for achieving Technical Conformance (PDF).
 
-    On approval of a use case, we will support you through the assurance process through to go live. We will discuss the assurance process and artefacts with you to help you understand the requirements.
+    ### Clinical assurance process
```
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:192]: Not sure about this clinical assurance section - has this been checked with CSO? We've said explicitly that the API isn't for clinical use, so why is clinical testing required?

File: specification/gp-connect-access-record-fhir.yaml (hunk 2 above)
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:255]: this needs to be a FHIR operation on a Patient resource, as per https://developer.nhs.uk/apis/gpconnect-1-6-0/accessrecord_structured_development_retrieve_patient_record.html#fhir-relative-request

```suggestion
  /FHIR/STU3/Patient/$access-record-structured:
```

File: specification/gp-connect-access-record-fhir.yaml [hunk 4]
```diff
@@ -1,13 +1,20 @@
 # This is an OpenAPI Specification (https://swagger.io/specification/)
 # for gp-connect-access-record-fhir owned by NHS Digital (https://digital.nhs.uk/)
 openapi: '3.0.0'
+x-spec-publication:
+  operation-order:
+    - operations:
+      - method: POST
+        path: /FHIR/STU3/$access-record-structured
```
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:8]: this needs to be an operation on /Patient/, consistent with endpoint below on lne 255
```suggestion
        path: /FHIR/STU3/Patient/$access-record-structured
```

File: specification/gp-connect-access-record-fhir.yaml [hunk 5]
```diff
@@ -33,79 +40,105 @@ info:
...
-
-    If you are confident that you can meet the prerequisites, please express an interest with the GP Connect team (see 'Onboarding' below).
+    * have a clinical safety officer (CSO) who is responsible for DCB0129 and, if necessary, DCB0160. For more on clinical risk management, visit Clinical risk management standards
 
     ## Related APIs
-
+    
     The following APIs are related to this API:
+    
+    * [GP Connect Access Document - FHIR API](https://digital.nhs.uk/developer/api-catalogue/gp-connect-access-document-fhir) - use this API to retrieve documents from a patient’s registered GP practice, after you've found the relevant document references within Access Record returns, use GP Connect Access Document to retrieve the documents. All documents can be requested separately.
```
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:77]: Don't think this is relevant as can't be used for patient facing services

**Review by agreenhalgh — CHANGES_REQUESTED** [review]
See comments on individual parts, main ussues are that we need to update to use an operation on /Patient/ and to make sure environments and servers sections are consistent

File: specification/gp-connect-access-record-fhir.yaml [hunk 6]
```diff
@@ -213,4 +262,119 @@ servers:
...
+                      valueBoolean: true
+                    - name: includeStatus
+                      valueBoolean: true
+                - name: includeUncategorisedData
+                - name: includeInvestigations
+                - name: includeReferrals
+                - name: includeMedication
+      responses: 
+        '201':
+          description: Information surrounding how the responses are returned can be found [here.](https://simplifier.net/guide/pfs-implementation-guide/Home/FHIR-Assets/Profiles.page.md?version=current)
```
[the remark this answers is not in the document]
Comment (edtu1) [reply specification/gp-connect-access-record-fhir.yaml:314]: updated to https://simplifier.net/guide/pfs-implementation-guide?version=current

File: specification/gp-connect-access-record-fhir.yaml [hunk 7]
```diff
@@ -213,4 +262,119 @@ servers:
...
+      name: Authorization
+      description: |
+        An [OAuth 2.0 bearer token](https://digital.nhs.uk/developer/guides-and-documentation/security-and-authorisation#user-restricted-apis).
+        Required in all environments except sandbox.
+      required: true
+      schema:
+        type: string
+        format: '^Bearer\ [[:ascii:]]+$'
+        example: 'Bearer g1112R_ccQ1Ebbb4gtHBP1aaaNM'
+    RoleId:
```
[the remark this answers is not in the document]
Comment (edtu1) [reply specification/gp-connect-access-record-fhir.yaml:329]: Done

File: specification/gp-connect-access-record-fhir.yaml [hunk 8]
```diff
@@ -33,79 +40,105 @@ info:
...
     - write back to the GP record
 
-    ## Retrieving documents
-
-    To retrieve documents from a patient’s registered GP practice, use [GP Connect Access Document - FHIR API](https://digital.nhs.uk/developer/api-catalogue/gp-connect-access-document-fhir). After you've found the relevant document references within Access Record returns, use GP Connect Access Document to retrieve the documents. All documents can be requested separately.
-
     ## Who can use this API
 
-    This API can be used by developers of person facing systems that support citizens gaining access to their GP practice medical record.
+    This API can be used by developers of clinical systems that support clinicans delivering direct care. It is a patient-facing API.
```
[the remark this answers is not in the document]
Comment (edtu1) [reply specification/gp-connect-access-record-fhir.yaml:45]: Updated.

File: specification/gp-connect-access-record-fhir.yaml (hunk 5 above)
Comment (edtu1) [reply specification/gp-connect-access-record-fhir.yaml:77]: removed

**Review by agreenhalgh — COMMENTED** [review]
still need to sort out some server paths but will raise seperately

File: specification/gp-connect-access-record-fhir.yaml [hunk 9]
```diff
@@ -33,184 +40,264 @@ info:
...
     description: Sandbox environment.
-  - url: 'https://int.api.service.nhs.uk/gp-connect-access-record-fhir'
+  - url: 'https://int.api.service.nhs.uk/gp-connect/patient-facing/get-structured-record'
     description: Integration test environment.
-  - url: 'https://api.service.nhs.uk/gp-connect-access-record-fhir'
+  - url: 'https://api.service.nhs.uk/gp-connect/patient-facing/get-structured-record'
     description: Production environment.
-paths: {}
+paths: 
+  /FHIR/STU3/Patient/$access-record-structured:
```
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:201]: This is the same operation as access record structured in GPC - if it's the same operation then the operation name needs to be consistent.

```suggestion
  /FHIR/STU3/Patient/$gpc.getstructuredrecord
```

File: specification/gp-connect-access-record-fhir.yaml [hunk 10]
```diff
@@ -1,13 +1,20 @@
 # This is an OpenAPI Specification (https://swagger.io/specification/)
 # for gp-connect-access-record-fhir owned by NHS Digital (https://digital.nhs.uk/)
 openapi: '3.0.0'
+x-spec-publication:
+  operation-order:
+    - operations:
+      - method: POST
+        path: /FHIR/STU3/Patient/$access-record-structured
```
Comment (agreenhalgh) [inline specification/gp-connect-access-record-fhir.yaml:8]: This is the same operation as access record structured in GPC - if it's the same operation then the operation name needs to be consistent.

```suggestion
  /FHIR/STU3/Patient/$gpc.getstructuredrecord
```

**Review by agreenhalgh — CHANGES_REQUESTED** [review]

**Review by agreenhalgh — APPROVED** [review]
LGTM

**Review by agreenhalgh — DISMISSED** [review]
Typo corrected

**Review by agreenhalgh — APPROVED** [review]

**Pull request merged**